File: backend/app/models/concept.py
from sqlalchemy import Column, Integer, String, ForeignKey, Text
from sqlalchemy.orm import relationship
from app.db.base_class import Base

class Concept(Base):
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, index=True)
    description = Column(Text)
    
    session_id = Column(Integer, ForeignKey("session.id"))
    session = relationship("Session", back_populates="concepts")

File: backend/models/Concept.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Concept = sequelize.define('Concept', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },

  name: {
    type: DataTypes.STRING
  },

  description: {
    type: DataTypes.TEXT
  },

  session_id: {
    type: DataTypes.INTEGER
  }
}, {
  tableName: 'concepts',
  timestamps: false
});

module.exports = Concept;
